modules/rostests/apitests/shell32/ShellExecuteEx.cpp

```diff
@@ -111,15 +111,37 @@ getCommandLineFromProcess(HANDLE hProcess)
     PEB peb;
     PROCESS_BASIC_INFORMATION info;
     RTL_USER_PROCESS_PARAMETERS Params;
+    LONG status;
+    BOOL ret;
+
+    status = NtQueryInformationProcess(hProcess, ProcessBasicInformation, &info, sizeof(info), NULL);
+    if (status != 0)
+        trace("status: 0x%08lX\n", status);
+
+    ret = ReadProcessMemory(hProcess, info.PebBaseAddress, &peb, sizeof(peb), NULL);
+    if (!ret)
+        trace("ReadProcessMemory failed (%ld)\n", GetLastError());
 
-    NtQueryInformationProcess(hProcess, ProcessBasicInformation, &info, sizeof(info), NULL);
-    ReadProcessMemory(hProcess, info.PebBaseAddress, &peb, sizeof(peb), NULL);
     ReadProcessMemory(hProcess, peb.ProcessParameters, &Params, sizeof(Params), NULL);
+    if (!ret)
+        trace("ReadProcessMemory failed (%ld)\n", GetLastError());
 
     LPWSTR cmdline = Params.CommandLine.Buffer;
+    if (!cmdline)
+        trace("!cmdline\n");
+
     SIZE_T cchCmdLine = Params.CommandLine.Length;
+    if (!cchCmdLine)
+        trace("!cchCmdLine\n");
+
     LPWSTR pszBuffer = (LPWSTR)calloc(cchCmdLine + 1, sizeof(WCHAR));
-    ReadProcessMemory(hProcess, cmdline, pszBuffer, cchCmdLine, NULL);
+    if (!pszBuffer)
+        trace("!pszBuffer\n");
+
+    ret = ReadProcessMemory(hProcess, cmdline, pszBuffer, cchCmdLine, NULL);
```

Comment:
The function is failed at here.

![VirtualBox_ReactOS_14_03_2024_09_06_16](https://github.com/reactos/reactos/assets/2107452/c4b50802-5a9c-46ab-a2d7-76d25783a3cf)